lib/server.go

```diff
@@ -148,6 +148,11 @@ type Server struct {
 	// timer is a helper variable that allows timing events for development purposes.
 	// It can be used to find computational bottlenecks.
 	timer *Timer
+
+	// DbMutex protects the badger database from concurrent access when it's being closed & re-opened.
+	// This is necessary because the database is closed & re-opened when the node finishes hypersyncing in order
+	// to change the database options from Default options to Performance options.
+	DbMutex deadlock.Mutex
```

Comment:
So we're doing a couple things here:
1. We reduce the hypersync queue size. Seems like a strict win.
2. We swap the db options on badger after we finish syncing. This one creates some messiness.

Is (2) a significant improvement to memory usage? Like if we merge (1) but not (2) does it go above 32gb?

Reply:
2 does win us a substantial amount of memory usage. The intense I/O requirements of hypersync married with our very high performance settings results in a massive amount of memory usage. 

An alternate approach here would be to fix the way we store utxo ops in the backend db. My hunch is that if we were to fix that index to only store a single utxo op per record, we would be able to run default settings in block sync. This would involve moving the block + transaction to the key of that index, rather than having a single record that contains a bundle with every op for every transaction in a block.